Index: xivo-python-swaggerpy-packaging/swaggerpy/http_client.py
===================================================================
--- xivo-python-swaggerpy-packaging.orig/swaggerpy/http_client.py
+++ xivo-python-swaggerpy-packaging/swaggerpy/http_client.py
@@ -197,10 +197,10 @@ class SynchronousHttpClient(HttpClient):
         url = preped_req.url
         # Requests version 2.0.0 (at least) will no longer form a URL for us
         # for ws scheme types, so we do it manually
-        if params:
-            joined_params = "&".join(["%s=%s" % (k, v)
-                                     for (k, v) in params.items()])
-            url += "?%s" % joined_params
+        #if params:
+        #    joined_params = "&".join(["%s=%s" % (k, v)
+        #                             for (k, v) in params.items()])
+        #    url += "?%s" % joined_params
         return websocket.create_connection(url, header=header)
 
     def apply_authentication(self, req):
